src/Kriti/Parser/Lexer.x

```diff
@@ -0,0 +1,133 @@
+{
+module Kriti.Parser.Lexer where
+
+import qualified Data.Text as T
+import Kriti.Parser.Monad
+import Kriti.Parser.Spans
+import Kriti.Parser.Token
+}
+
+$digit = 0-9
+$alpha = [a-zA-Z]
+$alphanum = [a-zA-Z09]
+
+tokens :-
+
+-- Whitespace insensitive
+<0, expr> $white+                 ;
+
+-- Comments
+<0, expr> "#".*                         ;
+
+-- Syntax
+<0> range                                      { token TokIdentifier }
+<0, expr> true                                 { token (TokBoolLit . (\(Loc sp _) -> Loc sp True)) }
+<0, expr> false                                { token (TokBoolLit . (\(Loc sp _) -> Loc sp False)) }
+<0, expr> \$                                   { token TokIdentifier }
+<0, expr> \$? $alpha [\$ $alpha $digit \_ \-]* { token TokIdentifier }
+
+-- | String Templating
```

Comment:
Template lexing is done using [Start Codes](https://www.haskell.org/alex/doc/html/alex-files.html#startcodesl). I tried to document this part of the lexer pretty heavily.